src/main/java/org/sefglobal/scholarx/controller/AuthUserController.java

```diff
@@ -37,6 +39,16 @@ public Object getLoggedUser(Authentication authentication)
     return authentication.getPrincipal();
   }
 
+  @PostMapping
+  @ResponseStatus(HttpStatus.OK)
+  public Profile updateUserDetails(
+          Authentication authentication,
+          @Valid @RequestBody Profile profile)
+          throws ResourceNotFoundException, UnauthorizedException {
+    Profile profile = (Profile) authentication.getPrincipal();
+    return profileService.updateUserDetails(profile.getUid, profile);
```

Comment:
There are 2 objects with the name `profile` here. Change the name of the object taken from the request body

Reply:
Also it should pass the `id` of the profile as the service method. Not the `uuid`